libcasr/src/lua.rs

```diff
@@ -0,0 +1,337 @@
+//! Lua module implements `ParseStacktrace`, `CrashLineExt` and `Severity` traits for Lua reports.
+use crate::error::*;
+use crate::execution_class::ExecutionClass;
+use crate::severity::Severity;
+use crate::stacktrace::{CrashLine, CrashLineExt, DebugInfo};
+use crate::stacktrace::{ParseStacktrace, Stacktrace, StacktraceEntry};
+
+use regex::Regex;
+
+/// Structure provides an interface for save parsing lua exception.
+#[derive(Clone, Debug)]
+pub struct LuaException {
+    message: String,
+}
+
+impl LuaException {
+    /// Create new `LuaException` instance from stream
+    pub fn new(stream: &str) -> Option<Self> {
+        let re = Regex::new(r#"\S+: .+\n\s*stack traceback:\s*\n(?:.*\n)*.+: .+"#).unwrap();
+        let mat = re.find(stream)?;
+        Some(LuaException {
+            message: mat.as_str().to_string(),
+        })
+    }
+    /// Extract stack trace from lua exception.
+    pub fn extract_stacktrace(&self) -> Result<Vec<String>> {
```

Comment:
For `stacktrace` and `exception` we have different interfaces. Why is this code here?

Reply:
The word `exception` in the interaface name can create confusion.
I just want to create additional layer of abstruction like `UbsanWarning` and use resulting interface for safe work with all necessary methods from `ParseStacktrace `, `CrashLineExt` and `Severity`.
Maybe it's worth to renaming it to 'LuaError' or etc.

Reply:
I think we should handle Lua errors the same way we handle errors in other supported languages. We should implement the `Exception` trait for `LuaError` like [here](https://github.com/ispras/casr/blob/af9e9e46df70e81b8d5275748863c3a4007faa5e/libcasr/src/rust.rs#L13)

Reply:
But it's already implement the `Severity` trait

Reply:
The main idea is transfer more parse logic to lib for having single entry point like for [`UbsanWarning`](https://github.com/ispras/casr/blob/af9e9e46df70e81b8d5275748863c3a4007faa5e/libcasr/src/ubsan.rs#L177).
In result you just give stream to lib and recieve all necessary info.